src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    this.CustomHttpHeaders = new Dictionary <string, string> ();
+
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');

```

Comment:
the value can have "}", right? how about it?

Reply:
We are trimming the last "}" so that we don't have an empty string at the end when we split the string via the remaining "}"